snykTask/src/install/index.ts

```diff
@@ -54,17 +54,51 @@ export async function downloadExecutable(
   targetDirectory: string,
   executable: Executable,
 ) {
-  const fileWriter = fs.createWriteStream(
-    path.join(targetDirectory, executable.filename),
-    {
-      mode: 0o766,
-    },
-  );
-  return new Promise<void>((resolve, reject) => {
-    https.get(executable.downloadUrl, (response) => {
-      response.on('end', () => resolve());
-      response.on('error', (err) => reject(err));
-      response.pipe(fileWriter);
-    });
+  const filePath = path.join(targetDirectory, executable.filename);
+
+  // Check if the file already exists
+  if (fs.existsSync(filePath)) {
```

Comment:
Question: Don't we need to compare the latest version and the current version?

Reply:
The files don't survive between pipeline runs so we don't need to worry about it